packages/shared/src/utils/convertToMarkdownLinks.ts

```diff
@@ -0,0 +1,8 @@
+// Regular expression to match URLs
+const urlRegex =
+  /(?:https:\/\/(?:www\.[^\s/$.?#].[^\s]*[^\s.,!?#$]|(?!www\.)[^\s/$.?#].[^\s]*[^\s.,!?#$])|www\.[^\s/$.?#].[^\s]*[^\s.,!?#$])/g;
+
+export const convertToMarkdownLinks = (text: string) => {
+  const unformattedStr = text.replace(/\[([^\]]+)\]\([^)]+\)/g, '$1');
+  return unformattedStr.replace(urlRegex, (url: string) => `[${url}](${url})`);
+};
```

Comment:
nit: could we rename this function to be `replaceUrlsWithMarkdownFormat`? otherwise the function sounds like it'll convert _any_ arbitrary string to markdown links